addons/website/models/res_company.py

```diff
@@ -19,6 +20,17 @@ def action_open_website_theme_selector(self):
         action['target'] = 'new'
         return action
 
+    @api.constrains('active')
+    def _check_active(self):
+        super()._check_active()
+        for company in self:
+            if not company.active:
+                public_users_count = self._get_public_users_count_for_website()
+                if public_users_count:
+                    raise ValidationError(_(
+                        'The company %(company_name)s cannot be archived because it has some public users', company_name=company.name
+                    ))
```

Comment:
It would be about right (except that it's not I think, see below I'll explain why), but it should also consider the case of having a website_id set on the company but no public user linked to this company (regardless if this is a legit flow or not, it wouldn't cost much?)

That part would just be about
```py
if company.website_id:
    # this company is linked to a website
```

And it's actually all it should check, as it will implicitly also ensure that the user doesn't keep a website `user_id` linked to that company. Indeed:
- When creating a website or modifying it, the code ensure the fact that a public user linked to the company of that website is set.
- So, if you change the company of a website, it will also change the user_id (public user field) of that website
- Then, if you prevent the archive of company if the company is linked to a website, it would mean that the user will have to change the website company, and doing so it will also change the public user.

At the end, unless I am missing something, just checking for `company.website_id:` will cover everything.

Reply:
@rdeodoo  Ok thanks for the explanation. I've simplified the commit.